Source/WebCore/Modules/url-pattern/URLPatternConstructorStringParser.cpp

```diff
@@ -0,0 +1,370 @@
+{
+    if (!isNonSpecialPatternChar(m_tokenIndex+1, '/'))
+        return false;
+    if (!isNonSpecialPatternChar(m_tokenIndex+2, '/'))
+        return false;
+    return true;
+}
+
+// https://urlpattern.spec.whatwg.org/#make-a-component-string
+String URLPatternConstructorStringParser::makeComponentString() const
+{
+    ASSERT(m_tokenIndex < m_tokenList.size());
+    const auto& token = m_tokenList[m_tokenIndex];
+
+    auto componentStartToken = getSafeToken(m_componentStart);
+    ASSERT(componentStartToken.index && token.index);
+    auto componentStartIndex = *componentStartToken.index;
+
+    return m_input.substring(componentStartIndex, *token.index - componentStartIndex).toString();
+}
+
+static inline void setInitComponentFromState(URLPatternInit& init, URLPatternConstructorStringParserState state, String componentString)
+{
+    switch (state) {
+    case URLPatternConstructorStringParserState::Protocol:
+        init.protocol = componentString;
+        break;
+    case URLPatternConstructorStringParserState::Username:
+        init.username = componentString;
+        break;
+    case URLPatternConstructorStringParserState::Password:
+        init.password = componentString;
+        break;
+    case URLPatternConstructorStringParserState::Hostname:
+        init.hostname = componentString;
+        break;
+    case URLPatternConstructorStringParserState::Port:
+        init.port = componentString;
+        break;
+    case URLPatternConstructorStringParserState::Pathname:
+        init.pathname = componentString;
+        break;
+    case URLPatternConstructorStringParserState::Search:
+        init.search = componentString;
+        break;
+    case URLPatternConstructorStringParserState::Hash:
+        init.hash = componentString;
+        break;
+    default:
+        break;
+    }
+}
+
+// https://urlpattern.spec.whatwg.org/#compute-protocol-matches-a-special-scheme-flag
+ExceptionOr<void> URLPatternConstructorStringParser::computeProtocolMatchSpecialSchemeFlag(ScriptExecutionContext& context)
+{
+    Ref vm = context.vm();
+    JSC::JSLockHolder lock(vm);
+
+    auto maybeProtocolComponent = URLPatternUtilities::URLPatternComponent::compile(vm, makeComponentString(), EncodingCallbackType::Protocol, URLPatternUtilities::URLPatternStringOptions { });
+    if (maybeProtocolComponent.hasException())
+        return maybeProtocolComponent.releaseException();
+
+    auto protocolComponent = maybeProtocolComponent.releaseReturnValue();
+    m_protocolMatchesSpecialSchemeFlag = protocolComponent.matchSpecialSchemeProtocol(context);
+
+    return { };
+}
+
+// https://urlpattern.spec.whatwg.org/#change-state
+void URLPatternConstructorStringParser::changeState(URLPatternConstructorStringParserState newState, size_t skip)
+{
+    if (m_state != URLPatternConstructorStringParserState::Init
+        && m_state != URLPatternConstructorStringParserState::Authority
+        && m_state != URLPatternConstructorStringParserState::Done)
```

Comment:
I *think* coding style says to use curly brackets if the condition is on multiple lines.

Reply:
You are right! I will change this

>  or a single statement spans multiple lines. 
<img width="931" alt="Screenshot 2024-12-19 at 12 07 45 PM" src="https://github.com/user-attachments/assets/536bac03-7942-49fa-9126-bfbd32e743f1" />